process/block/metablock.go

```diff
@@ -455,13 +469,20 @@ func (mp *metaProcessor) CommitBlock(
 		return err
 	}
 
+	numberOfShards := mp.shardCoordinator.NumberOfShards()
+	headersNonce := make([]uint64, numberOfShards, numberOfShards)
+
 	for i := 0; i < len(header.ShardInfo); i++ {
 		shardData := header.ShardInfo[i]
 		header, err := process.GetShardHeaderFromPool(shardData.HeaderHash, mp.dataPool.ShardHeaders())
 		if header == nil {
 			return err
 		}
 
+		if headersNonce[shardData.ShardId] < header.Nonce {
```

Comment:
this doesn't work well on blocks with nonce 0. Should change headersNonce to be a map instead of slice.

Reply:
Removed this code.